src/resources/views/base/inc/alerts.blade.php

```diff
@@ -1,21 +1,40 @@
 {{-- Bootstrap Notifications using Prologue Alerts & PNotify JS --}}
 <script type="text/javascript">
-  Noty.overrideDefaults({
-    layout   : 'topRight',
-    theme    : 'backstrap',
-    timeout  : 2500, 
-    closeWith: ['click', 'button'],
-  });
+    // This is intentionaly run after dom loads so this way we can avoid showing duplicate alerts
+    // when the user is beeing redirected by persistent table, that happens before this event triggers.
+    window.addEventListener('DOMContentLoaded', function() {
+        Noty.overrideDefaults({
+            layout: 'topRight',
+            theme: 'backstrap',
+            timeout: 2500,
+            closeWith: ['click', 'button'],
+        });
 
-  @foreach (\Alert::getMessages() as $type => $messages)
+        // get alerts from the alert bag
+        var $alerts_from_php = JSON.parse('@json(\Alert::getMessages())');
 
-      @foreach ($messages as $message)
+        // get the alerts from the localstorage
+        var $alerts_from_localstorage = JSON.parse(localStorage.getItem('backpack_alerts')) ?? {};
 
-        new Noty({
-          type: "{{ $type }}",
-          text: "{!! str_replace('"', "'", $message) !!}"
-        }).show();
+        // merge both php alerts and localstorage alerts
+        Object.entries($alerts_from_php).forEach(([type, messages]) => {
+            if(typeof $alerts_from_localstorage[type] !== 'undefined') {
+                $alerts_from_localstorage[type].push(...messages);
+            } else {
+                $alerts_from_localstorage[type] = messages;
+            }
```

Comment:
```suggestion
            $alerts_from_localstorage[type] ||= [];
            $alerts_from_localstorage[type].push(...messages);
```

Short-circuit evaluation, is it an option to make this code a little bit cleaner or is it too strange? 😅

Reply:
My main concern using this is that for example chrome, only supports it from 85+, which was launched Aug 2020. So, not even 1 year ago.

Reply:
@pxpm Agreed!
Now that you say that, I found this; https://caniuse.com/mdn-javascript_operators_logical_or_assignment
And even apple, once again, is a lot behind on this 😅

We could use the same version but without the short-circuit evaluation, but it would not be as much as efficient.
So let's move this refactor to the future 👍

```suggestion
            $alerts_from_localstorage[type] = $alerts_from_localstorage[type] || [];
            $alerts_from_localstorage[type].push(...messages);
```